### DISCUSSION TITLE
fastapi is waiting to response and there is no concurrency

### DISCUSSION THREAD

**Discussion by gpassing** [fastapi/fastapi#6021] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.websocket("/ws")
async def websocket_endpoint(websocket: WebSocket):
    await websocket.accept()
    while True:
        data = await websocket.receive_text()
        task= MyTask("eas.com", password="F")
        task.get_all_appointments()//this function take 1 min to done we can not wait it to do we need run similar request concurrently
        await websocket.send_text(f"Message text was: {data}")
```

### Description

(my task per request take a few minutes and we need handle many request concurrently)

suppose we had 100 request per seconds and each request takes a few minutes to create answer/response for user
but fastapi and uvicorn just run an event loop with one thread for all request then each request is waited to finish to process new request ...we want many request process in same time not one by one 

### Operating System

Windows

### FastAPI Version

0.89.1

### Python Version

3.10

**Comment by frankie567** [comment, marked as answer]
Yes, that's expected. Running a sync/blocking function in an `async` function will still block the event loop. If you want to prevent the process from blocking, you have two solutions:

1. (Preferred) It looks like `MyTask` is performing HTTP requests under the hood. If so, you should move to an async-compatible HTTP client, like [`httpx`](https://www.python-httpx.org/)
2. If it's not possible or if your function is CPU-bound (lot of heavy computations), try to call your function with `run_in_threadpool`. Its role is to schedule a blocking function into a background thread:

```py
from starlette.concurrency import run_in_threadpool 

await run_in_threadpool(task.get_all_appointments())
```
